apps/arena/lib/arena/game_updater.ex

```diff
@@ -1659,6 +1660,37 @@ defmodule Arena.GameUpdater do
     )
   end
 
+  defp random_position_in_square(object_radius, external_wall, obstacles, initial_position, square_wall) do
+    x =
+      Enum.random(trunc(square_wall.left)..trunc(square_wall.right)) / 1.0 + initial_position.x
+
+    y =
+      Enum.random(trunc(square_wall.bottom)..trunc(square_wall.top)) / 1.0 + initial_position.y
+
+    circle = %{
+      id: 1,
+      shape: :circle,
+      position: %{x: x, y: y},
+      radius: object_radius,
+      vertices: [],
+      speed: 0.0,
+      category: :power_up,
+      direction: %{x: 0.0, y: 0.0},
+      is_moving: false,
+      name: "Circle 1"
+    }
+
+    collisionable_obstacles =
+      Map.filter(obstacles, fn {_obstacle_id, obstacle} -> obstacle.aditional_info.collisionable end)
+
+    Physics.get_closest_available_position(
+      circle.position,
+      circle,
+      external_wall,
+      collisionable_obstacles
+    )
```

Comment:
Let's put this inside a function so both power ups and items use the same logic.

Reply:
done in [feat: modularize random position functions](https://github.com/lambdaclass/mirra_backend/pull/977/commits/218dfa85dc290beb4cb49997a605f67e80f368c7)